CoreUtilities/src/au/gov/asd/tac/constellation/utilities/svg/SVGData.java

```diff
@@ -0,0 +1,346 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+package au.gov.asd.tac.constellation.utilities.svg;
+
+import au.gov.asd.tac.constellation.utilities.text.SeparatorConstants;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Data structure to store SVG information and associations.
+ * Tree-like data structure capable of storing SVG types, attributes and children.
+ * acts as a bare-bones object to be able to store SVG Data for association with minimal impact on memory.
+ * 
+ * @author capricornunicorn123
+ */
+public class SVGData {
+
+    private static final Logger LOGGER = Logger.getLogger(SVGData.class.getName());
+    private final String type;
+    private final Map<String, String> attributes;
+    private final Map<String, SVGData> children;
+    private String content;
+    private SVGData parent;
+
+    public SVGData(final String type, final SVGData parent, final Map<String, String> attributes) {
+        if (type == null){
+            throw new NullPointerException("SVGData elements cannot have a type Null");
+        }
+        this.type = type;
+        this.children = new LinkedHashMap<>();
+        this.attributes = SVGAttributeConstant.initialiseBasicAttributes();
+        if (attributes != null) {
+            this.attributes.putAll(attributes);
+        } 
+        this.setParent(parent);
+        this.content = null;
+    }
+    
+    /**
+     * Returns the SVG type.
+     * @return 
+     */
+    public final String getType() {
+        return this.type;
+    }
+    
+    /**
+     * Adds a child SVGObject to the current SVG element.
+     * To avoid duplication, this method should only be called by the setParent() method.
+     * Child SVGObjects are stored in HashMap based on their id attribute.
+     * @param child 
+     */
+    private void setChild(final SVGData child) {
+        final String childID = child.getAttributeValue(SVGAttributeConstant.ID.getName());
+        this.children.put(childID, child);
+    }
+    
+    /**
+     * Returns an SVGData with a specified id attribute value.
+     * Use depth first search.
+     * @param idValue 
+     * @return child
+     */
+    public SVGData getChild(final String idValue) {
+        SVGData child = this.children.get(idValue);
+        if (child == null) {
+            for (SVGData childIndex : this.getAllChildren()) {
+                child = childIndex.getChild(idValue);
+                if (child != null) {
+                    return child;
```

Comment:
just noting the return here is the same as it is out of the loop. Perhaps just adding a break here is what is needed? (probably not essential but a thought)